atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java

```diff
@@ -262,28 +262,27 @@ public String toString() {
 
         poolConfig.setJmxNamePrefix(CassandraLogHelper.host(host));
         GenericObjectPool<Client> pool = new GenericObjectPool<>(cassandraClientFactory, poolConfig);
-        registerMetrics(pool, poolNumber);
+        registerMetrics(pool, host.getHostString());
         return pool;
     }
 
-    private void registerMetrics(GenericObjectPool<Client> pool, int poolNumber) {
-        registerMetric(getMetricName("meanActiveTimeMillis", poolNumber), pool::getMeanActiveTimeMillis);
-        registerMetric(getMetricName("meanIdleTimeMillis", poolNumber), pool::getMeanIdleTimeMillis);
-        registerMetric(getMetricName("meanBorrowWaitTimeMillis", poolNumber), pool::getMeanBorrowWaitTimeMillis);
-        registerMetric(getMetricName("numIdle", poolNumber), pool::getNumIdle);
-        registerMetric(getMetricName("numActive", poolNumber), pool::getNumActive);
-        registerMetric(getMetricName("approximatePoolSize", poolNumber), () -> pool.getNumIdle() + pool.getNumActive());
-        registerMetric(getMetricName("proportionDestroyedByEvictor", poolNumber),
+    private void registerMetrics(GenericObjectPool<Client> pool, String hostName) {
+        registerMetric("meanActiveTimeMillis", hostName, pool::getMeanActiveTimeMillis);
+        registerMetric("meanIdleTimeMillis", hostName, pool::getMeanIdleTimeMillis);
+        registerMetric("meanBorrowWaitTimeMillis", hostName, pool::getMeanBorrowWaitTimeMillis);
+        registerMetric("numIdle", hostName, pool::getNumIdle);
+        registerMetric("numActive", hostName, pool::getNumActive);
+        registerMetric("approximatePoolSize", hostName, () -> pool.getNumIdle() + pool.getNumActive());
+        registerMetric("proportionDestroyedByEvictor", hostName,
                 () -> ((double) pool.getDestroyedByEvictorCount()) / ((double) pool.getCreatedCount()));
-        registerMetric(getMetricName("proportionDestroyedByBorrower", poolNumber),
+        registerMetric("proportionDestroyedByBorrower", hostName,
                 () -> ((double) pool.getDestroyedByBorrowValidationCount()) / ((double) pool.getCreatedCount()));
     }
 
-    private String getMetricName(String metric, int poolNumber) {
-        return "pool" + poolNumber + "." + metric;
-    }
-
-    private void registerMetric(String metricName, Gauge gauge) {
-        metricsManager.registerMetric(CassandraClientPoolingContainer.class, metricName, gauge);
+    private void registerMetric(String metricName, String hostName, Gauge gauge) {
+        metricsManager.registerMetric(CassandraClientPoolingContainer.class,
+                metricName,
+                gauge,
+                ImmutableMap.of("cassandra-host", hostName));
```

Comment:
as per my email, I'd recommend holding of on tagging for the moment.

Reply:
following up here. I don't think we want to tag the host. we've removed similar tags from remoting already: https://github.com/palantir/http-remoting/pull/642